### PR TITLE
Update CHKC and CHKO Type

### PR DESCRIPTION
**Pull request by wurzka**
1. Generated new JSON schema automatically
2. Serialised interface with CHKC-type/CHKO-type
3. Serialised example class

### CODE REVIEW

File: file-formats/chko/chko.json [hunk 1]
```diff
@@ -1,112 +1,113 @@
...
+        "type": "object",
+        "properties": {
+          "technicalId": {
+            "title": "Technical ID",
+            "description": "The technical Id of a parameter",
+            "type": "string",
+            "maxLength": 4
+          },
+          "name": {
+            "title": "Parameter name",
```
Comment (albertmink) [inline file-formats/chko/chko.json:86]:
```suggestion
            "title": "Parameter Name",
```

File: file-formats/chko/chko.json [hunk 2]
```diff
@@ -1,112 +1,113 @@
...
+            "maxLength": 30
+          },
+          "description": {
+            "title": "Parameter Description",
+            "description": "Description of a parameter",
+            "type": "string",
+            "maxLength": 255
+          },
+          "modifiable": {
+            "title": "Modifiable flag",
```
Comment (albertmink) [inline file-formats/chko/chko.json:98]:
```suggestion
            "title": "Modifiable Flag",
```

**Review by albertmink — COMMENTED** [review]
Found some issues with Title Case for `title`

File: file-formats/chko/chko.json (hunk 2 above)
Comment (wurzka) [reply file-formats/chko/chko.json]: Done

File: file-formats/chko/chko.json (hunk 1 above)
Comment (wurzka) [reply file-formats/chko/chko.json]: Done

File: file-formats/chkc/type/zif_atc_aff_chkc_v1.intf.abap [hunk 3]
```diff
@@ -0,0 +1,21 @@
+INTERFACE zif_atc_aff_chkc_v1
+  PUBLIC.
+
+  TYPES:
+    "! <p class="shorttext">Object Type CHKC</p>
+    "! CHKC object type
```
Comment (schneidermic0) [inline file-formats/chkc/type/zif_atc_aff_chkc_v1.intf.abap:6]: Small suggestion to replace CHKC by the name of the object type "ATC check category"

```suggestion
    "! <p class="shorttext">ATC Check Category</p>
    "! ATC an check category properties
```

File: file-formats/chkc/type/zif_atc_aff_chkc_v1.intf.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+  "$schema": "https://github.com/SAP/abap-file-formats/blob/main/file-formats/intf/intf.json",
+  "header": {
+    "description": "CHKO AFF Types",
```
Comment (schneidermic0) [inline file-formats/chkc/type/zif_atc_aff_chkc_v1.intf.json:4]:
```suggestion
    "description": "CHKC AFF Types",
```

File: file-formats/chko/type/zif_atc_aff_chko_v1.intf.abap [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+  TYPES:
+    "! <p class="shorttext">Parameters</p>
+    "! List of parameters
+    ty_parameters TYPE HASHED TABLE OF ty_parameter WITH UNIQUE KEY technical_id
+      WITH UNIQUE HASHED KEY xml_name COMPONENTS name.
+
+  TYPES:
+    "! <p class="shorttext">Object Type CHKO</p>
+    "! CHKO object type
```
Comment (schneidermic0) [inline file-formats/chko/type/zif_atc_aff_chko_v1.intf.abap:30]:
```suggestion
    "! <p class="shorttext">ATC Check</p>
    "! ATC check object properties
```

File: file-formats/chko/type/zif_atc_aff_chko_v1.intf.abap [hunk 6]
```diff
@@ -0,0 +1,54 @@
+INTERFACE zif_atc_aff_chko_v1
+  PUBLIC.
+
+  TYPES:
+    "! <p class="shorttext">Parameter</p>
+    "! A parameter of an ATC Check Object
+    BEGIN OF ty_parameter,
+      "! <p class="shorttext">Technical ID</p>
+      "! The technical Id of a parameter
+      technical_id TYPE c LENGTH 4,
```
Comment (schneidermic0) [inline file-formats/chko/type/zif_atc_aff_chko_v1.intf.abap:10]: I am not sure, whether this is really needed

**Review by schneidermic0 — CHANGES_REQUESTED** [review]
Thanks. :) 
See my comments

File: file-formats/chkc/type/zif_atc_aff_chkc_v1.intf.abap (hunk 3 above)
Comment (wurzka) [reply file-formats/chkc/type/zif_atc_aff_chkc_v1.intf.abap:6]: Done.

File: file-formats/chkc/type/zif_atc_aff_chkc_v1.intf.json (hunk 4 above)
Comment (wurzka) [reply file-formats/chkc/type/zif_atc_aff_chkc_v1.intf.json:4]: Done.

File: file-formats/chko/type/zif_atc_aff_chko_v1.intf.abap (hunk 5 above)
Comment (wurzka) [reply file-formats/chko/type/zif_atc_aff_chko_v1.intf.abap:30]: Done.

File: file-formats/chko/type/zif_atc_aff_chko_v1.intf.abap (hunk 6 above)
Comment (wurzka) [reply file-formats/chko/type/zif_atc_aff_chko_v1.intf.abap:10]: We use a type which is a hashed table with unique key technical_id. As the persistence class transfers our type into this type, there should be this component.

File: file-formats/chkc/type/zif_atc_aff_chkc_v1.intf.abap (hunk 3 above)
Comment (wurzka) [reply file-formats/chkc/type/zif_atc_aff_chkc_v1.intf.abap:6]: Is the 'an' in your description intentional? I removed it in my commit. But I can add it, if it should be there.

Comment (schneidermic0) [reply file-formats/chkc/type/zif_atc_aff_chkc_v1.intf.abap]: Thanks, without 'an' is fine

File: file-formats/chko/type/zif_atc_aff_chko_v1.intf.abap [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+      schema             TYPE string,
+      "! <p class="shorttext">Header</p>
+      "! header
+      "! $required
+      header             TYPE zif_aff_types_v1=>ty_header_60,
+      "! <p class="shorttext">Category</p>
+      "! The parent category of an ATC Check Object
+      category           TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Implementing Class</p>
+      "! The implementing class of the ATC Check Object
```
Comment (schneidermic0) [inline file-formats/chko/type/zif_atc_aff_chko_v1.intf.abap:44]:
```suggestion
      "! The implementing class of the ATC check object
```

File: file-formats/chko/type/zif_atc_aff_chko_v1.intf.abap [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  TYPES:
+    "! <p class="shorttext">ATC Check</p>
+    "! ATC check object properties
+    BEGIN OF ty_main,
+      "! <p class="shorttext">Schema</p>
+      "! Format version
+      "! $required
+      schema             TYPE string,
+      "! <p class="shorttext">Header</p>
+      "! header
```
Comment (schneidermic0) [inline file-formats/chko/type/zif_atc_aff_chko_v1.intf.abap:37]:
```suggestion
      "! Header
```

File: file-formats/chko/type/zif_atc_aff_chko_v1.intf.abap [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+      "! $required
+      header             TYPE zif_aff_types_v1=>ty_header_60,
+      "! <p class="shorttext">Category</p>
+      "! The parent category of an ATC Check Object
```
Comment (schneidermic0) [inline file-formats/chko/type/zif_atc_aff_chko_v1.intf.abap:41]: The category is a required field for an ATC check

```suggestion
      "! The parent category of an ATC check object
      "! $required
```

**Review by schneidermic0 — CHANGES_REQUESTED** [review]
I think the field `category` should be marked as `$required`
Sorry, I found some small typos I missed in the first review.

File: file-formats/chko/type/zif_atc_aff_chko_v1.intf.abap [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+      category           TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Implementing Class</p>
+      "! The implementing class of the ATC Check Object
+      "! $required
+      implementing_class TYPE zif_aff_types_v1=>ty_object_name_30,
+      "! <p class="shorttext">Remote-enabled flag</p>
+      "! The ATC Check Object is remote-enabled
```
Comment (wurzka) [inline file-formats/chko/type/zif_atc_aff_chko_v1.intf.abap:48]: @schneidermic0 I guess here (and in the next description) it should be also ATC check object, shouldn't it? So we write it to lower case if it's in the description

File: file-formats/chko/type/zif_atc_aff_chko_v1.intf.abap (hunk 7 above)
Comment (wurzka) [reply file-formats/chko/type/zif_atc_aff_chko_v1.intf.abap]: Done.

File: file-formats/chko/type/zif_atc_aff_chko_v1.intf.abap (hunk 9 above)
Comment (wurzka) [reply file-formats/chko/type/zif_atc_aff_chko_v1.intf.abap]: Done.

**Review by schneidermic0 — APPROVED** [review]
LGTM

**Pull request merged**